modules/userId/index.js

```diff
@@ -481,15 +482,40 @@ function getCombinedSubmoduleIdsForBidder(submodules, bidder) {
   if (!Array.isArray(submodules) || !submodules.length || !bidder) {
     return {};
   }
-  return submodules
+  const eligibleSubmodules = submodules
     .filter(i => !i.config.bidders || !isArray(i.config.bidders) || includes(i.config.bidders, bidder))
+
+  return getPrioritizedCombinedSubmoduleIds(eligibleSubmodules);
+}
+
+/**
+ * @param {SubmoduleContainer[]} submodules
+ */
+function getPrioritizedCombinedSubmoduleIds(submodules) {
+  const combinedIdStates = submodules
     .filter(i => isPlainObject(i.idObj) && Object.keys(i.idObj).length)
     .reduce((carry, i) => {
       Object.keys(i.idObj).forEach(key => {
-        carry[key] = i.idObj[key];
+        const currentIdPriorityByName = idPriority[key]?.reverse().indexOf(i.submodule.name);
+        const currentIdPriorityByAlias = idPriority[key]?.reverse().indexOf(i.submodule.aliasName);
+        // eslint-disable-next-line
+        const currentIdPriority = currentIdPriorityByName ? currentIdPriorityByName : currentIdPriorityByAlias ? currentIdPriorityByAlias : - 1
```

Comment:
I don't think this works (and I don't see it in a test case) - assuming you meant "use the name first, then the alias". 
Because of how boolean casting works, this will always use the name (every value except `0` is truthy),  except when it was the last element in the priority array (reversed becomes `0`), in which case it will always use the alias, even if it was not in the array.

I am not sure that considering aliases here is necessary, but if we do, my approach would be to "resolve" them when the id priority is passed to `setConfig`; that would also help with performance. By that I mean, instead of just taking in `idPriority` verbatim from user input, do some processing so that when you get here `idPriority[key]` is already reversed, and does not contain any aliases - only the proper names.

Reply:
@dgirardi Good catch! Addressed both concerns - flipping priority and alias handling in https://github.com/prebid/Prebid.js/pull/9896/commits/edb04989e8d9063c7a4c99410a7b308c214688b8.